### PR TITLE
Accept `ESID` as logout result

### PR DESCRIPTION
**Pull request by sergiohs84**
Also removes 2 warnings: unused variables and comparission between
signed and unsigned variables.

### CODE REVIEW

**Comment by sergiohs84** [comment]
Do not merge it yet. The `ESID` result can be received at HTTP request level, instead of at Command level, when there's multiple commands in the HTTP request.

**Comment by sergiohs84** [comment]
We might consider whether MegaApi should return API_OK when the result of a logout is ESID??

File: src/commands.cpp [hunk 1]
```diff
@@ -1161,7 +1161,7 @@ void CommandLogout::procresult()
 {
     error e = (error)client->json.getint();
     MegaApp *app = client->app;
-    if(!e)
+    if(!e || e == API_ESID)
```
Comment (javiserrano) [inline src/commands.cpp]: I think that API_ESID can't be received as a command-level response, only as a request-level response.

File: src/megaapi_impl.cpp [hunk 2]
```diff
@@ -7295,6 +7295,15 @@ void MegaApiImpl::request_error(error e)
         request->setText(client->sslfakeissuer.c_str());
     }
 
+    if (e == API_ESID)
+    {
+        if (client->sctable)
+        {
+            client->sctable->remove();
+        }
```
Comment (javiserrano) [inline src/megaapi_impl.cpp]: I think that the cache of syncs should be also removed here like in `CommandLogout::procresult`

File: src/megaapi_impl.cpp [hunk 3]
```diff
@@ -7336,7 +7345,7 @@ void MegaApiImpl::logout_result(error e)
     MegaRequestPrivate* request = requestMap.at(client->restag);
     if(!request || (request->getType() != MegaRequest::TYPE_LOGOUT)) return;
 
-    if(!e)
+    if(!e || e == API_ESID)
```
Comment (javiserrano) [inline src/megaapi_impl.cpp]: I have some doutbs in this case, but I think that it's not possible to receive API_ESID here.

File: src/commands.cpp (hunk 1 above)
Comment (sergiohs84) [inline src/commands.cpp]: You're right, only at HTTP level.

File: src/megaapi_impl.cpp (hunk 3 above)
Comment (sergiohs84) [inline src/megaapi_impl.cpp]: Since the command doesn't receive the ESID and the SDK doesn't call `logout_result()` with that error, I agree with you.

**Pull request merged**